test/api/unit_roles_test.rb

```diff
@@ -46,4 +62,68 @@ def test_post_unit_roles_not_unique
     assert_equal to_post[:unit_id], last_response_body['unit_id']
     assert_equal to_post[:user_id], last_response_body['user_id']
   end
+
+  # PUT tests
+  # Replace a unit role
+  def test_put_unit_role
+    # Details to replace
+    data_to_put = {
+      unit_role: FactoryGirl.build(:unit_role),
+      auth_token: auth_token
+    }
+
+    # Update unit_role with id = 1
+    put_json '/api/unit_roles/1', data_to_put
+    
+    # Check if the PUT get through
+    assert_equal 200, last_response.status
+    
+    # Check if the details replaced match as expected
+    response_keys = %w(unit_id user_id)
+    first_unit_role = UnitRole.first
+    assert_json_matches_model(last_response_body, first_unit_role, response_keys)
```

Comment:
This may be an interesting one to review - what should we be able to change? I think in this case we cannot actually change anything. So perhaps put should not exist. You would have to review the front end code to check this - but I am pretty sure that we do not use put. So it should be gone from the backend entirely.